sys/arm64/arm64/trap.c

```diff
@@ -168,16 +168,17 @@ extern uint32_t generic_bs_poke_1f, generic_bs_poke_2f;
 extern uint32_t generic_bs_poke_4f, generic_bs_poke_8f;
 
 static bool
-test_bs_fault(void *addr)
+test_bs_fault(uintcap_t addr)
 {
-	return (addr == &generic_bs_peek_1f ||
-	    addr == &generic_bs_peek_2f ||
-	    addr == &generic_bs_peek_4f ||
-	    addr == &generic_bs_peek_8f ||
-	    addr == &generic_bs_poke_1f ||
-	    addr == &generic_bs_poke_2f ||
-	    addr == &generic_bs_poke_4f ||
-	    addr == &generic_bs_poke_8f);
+	/* TODO: Will need fixing for a purecap kernel */
```

Comment:
How will this need fixing for purecap?  It seems like it will work fine for purecap?

Reply:
Function pointer address will have the LSB set, but ELR will not even if trapping from C64, that info goes in SPSR.C64 only.